src/ORM/FieldType/DBComposite.php

```diff
@@ -240,8 +240,9 @@ public function setField($field, $value, $markChanged = true)
     {
         $this->objCacheClear();
 
-        // Skip non-db fields
+        // Non-db fields get assigned as normal properties
         if (!$this->hasField($field)) {
+            $this->$field = $value;
```

Comment:
I really dislike this change, as it encourages poor style. You shouldn't be using setField() to shiv in public access to potentially protected members, as it violates the open closed principle. ViewableData suffers from being an overly-generic "god" parent class in much the way that Object once was, but I don't think we should be shuffling it's behaviour into subclasses.

If you have a service you need injected, you should be implementing service endpoint getters and setters instead for those instead.

E.g.

```php
public function setMyService($service) {
	$this->service = $service
};
```

Not

```php
protected $myService; // protected, but will be set by injector anyway as though it was public
```